src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -170,7 +163,8 @@
   [rules :- Rules
    request :- ring/Request
    name :- schema/Str]
-  (if-let [ { matched-rule :rule matches :matches } (some #(match? % request) rules)]
+  (if-let [{matched-rule :rule matches :matches}
+           (some #(match? % request) (sort-rules rules))]
```

Comment:
Thanks for doing that.  I think of the sorting as being one of the aspects of "normalization" that happens anytime we need to load/convert data from its external to its internal representation.  Ignoring the performance implications, it would seem artificial to me to only do "some" of that normalization at load time vs. other parts at execution time.  I'd have a really hard time envisioning that we'd ever want to do a resort after initial rule load/reload.  So, ultimately, I think this is the best way to go.